### PR TITLE
KOGITO-1546 XStreamUtils kie-soup refactoring

### PR DESCRIPTION
**Pull request by tarilabs**
a suite of PRs:

### drools.git "v7x" stream
[kie-soup] this one
[droolsjbpm-build-bootstrap] https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1286
[appformer] https://github.com/kiegroup/appformer/pull/953
[drools] https://github.com/kiegroup/drools/pull/2874
[jbpm] https://github.com/kiegroup/jbpm/pull/1640
[droolsjbpm-integration] https://github.com/kiegroup/droolsjbpm-integration/pull/2078
[kie-wb-common] https://github.com/kiegroup/kie-wb-common/pull/3288
[drools-wb] https://github.com/kiegroup/drools-wb/pull/1356
[optaplanner-wb] https://github.com/kiegroup/optaplanner-wb/pull/367
[jbpm-wb] https://github.com/kiegroup/jbpm-wb/pull/1430
[kie-wb-playground] https://github.com/kiegroup/kie-wb-playground/pull/86
[droolsjbpm-tools] https://github.com/kiegroup/droolsjbpm-tools/pull/104

1. ALL and each of these PRs ^ have been **successfully** compiled locally before raising the respective PR.
2. These PRs ^ will need be merged at-once.

### Kogito stream
[kogito-runtimes] https://github.com/kiegroup/kogito-runtimes/pull/481

the Kogito stream ^ will need to await a community or tag-release, but it is **not** required to be merged at-once.

### CODE REVIEW

File: kie-soup-commons/pom.xml [hunk 1]
```diff
@@ -46,11 +46,6 @@
       <artifactId>cdi-api</artifactId>
       <scope>provided</scope>
     </dependency>
-    <dependency>
-      <groupId>com.thoughtworks.xstream</groupId>
-      <artifactId>xstream</artifactId>
-      <scope>provided</scope>
```
Comment (tarilabs) [inline kie-soup-commons/pom.xml:52]: moved into `kie-soup-xstream`

File: kie-soup-commons/pom.xml [hunk 2]
```diff
@@ -81,7 +76,6 @@
               org.kie.soup.commons.*
             </Export-Package>
             <Import-Package>
-              com.thoughtworks.xstream.*;resolution:=optional,
```
Comment (tarilabs) [inline kie-soup-commons/pom.xml:84]: moved into `kie-soup-xstream`

File: kie-soup-xstream/pom.xml [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    <java.module.name>org.kie.soup.xstream</java.module.name>
+  </properties>
+
+  <dependencies>
+
+    <!-- NOTE TO MAINTAINERS: This module is used by UberFire's VFS, which must remain independent of "web" technologies.
+         Don't add dependencies from this module to any other UberFire or Errai module! See UF-131 and UF-135 for details. -->
+
+    <dependency>
+      <groupId>com.thoughtworks.xstream</groupId>
```
Comment (tarilabs) [inline kie-soup-xstream/pom.xml:45]: no other dependency than XStream

**Comment by MarianMacik** [comment]
Please don't forget to run FDB once this is done.

**Comment by tarilabs** [comment]
> Please don't forget to run FDB once this is done

@MarianMacik either the decision will be determined by FDB here, or either in each of the PR, 

We cannot let the underterministic test fail compound for each of the related PR.

Do we agree?

**Comment by tarilabs** [comment]
Jenkins execute full downstream build

**Comment by MarianMacik** [comment]
> > Please don't forget to run FDB once this is done
> 
> @MarianMacik either the decision will be determined by FDB here, or either in each of the PR,
> 
> We cannot let the underterministic test fail compound for each of the related PR.

Sure, but by running FDB you easily check if you didn't forget something.
> Do we agree?

Yep, undeterministic tests can be ignored if it is obvious that they don't have anything to do with this PR.

**Comment by tarilabs** [comment]
> but by running FDB you easily check if you didn't forget something

Considering the scope of the change, maybe FDC would have been more appropriate ;)

**Comment by MarianMacik** [comment]
> Considering the scope of the change, maybe FDC would have been more appropriate ;)

Ah, you're right.

**Comment by tarilabs** [comment]
Jenkins execute full downstream compile

**Comment by MarianMacik** [comment]
@tarilabs I think it is
jenkins execute compile downstream build

**Comment by tarilabs** [comment]
Let's try ...

jenkins execute compile downstream build

**Comment by tarilabs** [comment]
Jenkins execute full downstream build

**Comment by tarilabs** [comment]
```
Test Result (7 failures / -48)
org.kie.workbench.common.dmn.client.common.BoxedExpressionHelperTest.org.kie.workbench.common.dmn.client.common.BoxedExpressionHelperTest
org.kie.wb.selenium.ui.LoadAllPerspectivesIntegrationTest.org.kie.wb.selenium.ui.LoadAllPerspectivesIntegrationTest
org.kie.wb.selenium.ui.LoginIntegrationTest.org.kie.wb.selenium.ui.LoginIntegrationTest
org.kie.wb.selenium.ui.ProjectLibraryIntegrationTest.org.kie.wb.selenium.ui.ProjectLibraryIntegrationTest
org.kie.wb.selenium.ui.StandalonePerspectivesIntegrationTest.org.kie.wb.selenium.ui.StandalonePerspectivesIntegrationTest
org.drools.eclipse.editors.rete.ReteooLayoutFactoryTest.testLayoutRowList
org.drools.eclipse.editors.rete.ReteooLayoutFactoryTest.testCalculateReteRows
```

I would argue un-related test failures

**Comment by MarianMacik** [comment]
Yes, and the compile build passed as well.

**Pull request merged**